indexing-solr/src/test/java/info/rmapproject/indexing/solr/repository/DiscoRepositoryTest.java

```diff
@@ -0,0 +1,461 @@
+        //ieee should hit all records
+		String search="(*ieee*)";
+		Pageable pageable = PageRequest.of(0, 10);
+				
+		//search for case insensitive string in additional statements
+		HighlightPage<DiscoSolrDocument> discos 
+				= discoRepository.findDiscoSolrDocumentsUsingRelatedStmtsAndHighlight(search, FILTER_ACTIVE, pageable);
+		//should only be 2 if active only
+		assertEquals(2, discos.getTotalElements());		
+		//make sure it's highlighting related statements
+		assertEquals("disco_related_statements", discos.getHighlighted().get(0).getHighlights().get(0).getField().toString());
+		//make sure highlight contains the search string
+		assertTrue(discos.getHighlighted().get(0).getHighlights().get(0).getSnipplets().get(0).toLowerCase().contains("ieee"));
+		//make sure it matches one discos
+		String discoId = discos.getContent().get(0).getDiscoUri();
+		assertTrue(discoId.equals(discoUri1) || discoId.equals(discoUri2));
+				
+		//search for case insensitive string in additional statements
+		discos = discoRepository.findDiscoSolrDocumentsUsingRelatedStmtsAndHighlight(search, FILTER_INACTIVE, pageable);
+		//should only be 1 if inactive only
+		discoId = discos.getContent().get(0).getDiscoUri();
+		assertEquals(1, discos.getTotalElements());		
+		assertEquals(discoUri3, discoId);		
+		
+		//search for case insensitive string in additional statements
+		discos = discoRepository.findDiscoSolrDocumentsUsingRelatedStmtsAndHighlight(search, FILTER_ACTIVEORINACTIVE, pageable);
+		//should only be 3 if all statuses
+		assertEquals(3, discos.getTotalElements());					
+	}
+	
+	
+	
+	/**
+	 * Creates and indexes a DiSCO based on TestFile reference provided
+	 * @param testFile
+	 * @return
+	 * @throws Exception
+	 */
+	private RMapEvent indexCreateDisco(TestFile testFile) throws Exception {
+		ORMapDiSCO disco1 = TestUtils.getRMapDiSCOObj(testFile);
+		String discoUri1 = disco1.getId().toString();
+        assertNotNull(discoUri1);
+		RMapEvent event = rmapService.createDiSCO(disco1, new RequestEventDetails(this.sysagent.getId().getIri()));
+        IndexDTO indexDto = new IndexDTO(event, this.sysagent, null, disco1);
+        discosIndexer.index(mapper.apply(indexDto));
+        return event;
+	}
+
+	/**
+	 * Inactivates a DiSCO and updates the index to reflect change.
+	 * @param discoUri
+	 * @return RMapEvent
+	 * @throws Exception
+	 */
+	private RMapEvent indexInactivateDisco(URI discoUri) throws Exception {
+		RMapEvent event = rmapService.inactivateDiSCO(discoUri, new RequestEventDetails(this.sysagent.getId().getIri()));
+		RMapDiSCO iDisco = rmapService.readDiSCO(discoUri);
+        IndexDTO indexDto = new IndexDTO(event, this.sysagent, iDisco, iDisco);
+        discosIndexer.index(mapper.apply(indexDto));
+        return event;
+	}
+
+	/**
+	 * Deletes a DiSCO and updates the index to reflect change.
+	 * @param discoUri
+	 * @return
+	 * @throws Exception
+	 */
+	private RMapEvent indexTombstoneDisco(URI discoUri) throws Exception {

```

Comment:
Note that the tombstoning logic has been  updated in PR #190

Reply:
I don't think that will make the test that uses this fail since it's only a 1 version test, but will check this after